app/controllers/references_controller.rb

```diff
@@ -1,16 +1,13 @@
 class ReferencesController < ApplicationController
-  before_filter :find_project
-  before_filter :authorize_deployer!
+  include CurrentProject
+
+  before_action :authorize_deployer!
```

Comment:
Yep, as I mentioned above, this is not a project-specific role check. That has been added in the PR about project-specific deployer role.